src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -290,23 +290,73 @@ void Instance::HandleAddOrUpdateWiFiNetwork(HandlerContext & ctx, const Commands
 {
     MATTER_TRACE_EVENT_SCOPE("HandleAddOrUpdateWiFiNetwork", "NetworkCommissioning");
     Commands::NetworkConfigResponse::Type response;
-    response.networkingStatus = ToClusterObjectEnum(mpDriver.Get<WiFiDriver *>()->AddOrUpdateNetwork(req.ssid, req.credentials));
+    MutableCharSpan debugText;
+#if CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER
+    char debugTextBuffer[CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER];
+    debugText = MutableCharSpan(debugTextBuffer);
+#endif
+    uint8_t outNetworkIndex   = 0;
+    response.networkingStatus = ToClusterObjectEnum(
+        mpDriver.Get<WiFiDriver *>()->AddOrUpdateNetwork(req.ssid, req.credentials, debugText, &outNetworkIndex));
+
+    size_t debugTextLen = strnlen(debugText.data(), debugText.size());
```

Comment:
Maybe let `AddOrUpdateNetwork` take `MutableCharSpan&` and be responsible for returning the correct size?

Reply:
Yes, exactly.

Especially because this code is buggy: it computes the strnlen, but then ignores it and dumps the whole buffer in the TLV anyway.